arangod/VocBase/Methods/Collections.cpp

```diff
@@ -481,12 +477,10 @@ Result Collections::unload(TRI_vocbase_t* vocbase, LogicalCollection* coll) {
 Result Collections::properties(Context& ctxt, VPackBuilder& builder) {
   LogicalCollection* coll = ctxt.coll();
   TRI_ASSERT(coll != nullptr);
-  ExecContext const* exec = ExecContext::CURRENT;
-  if (exec != nullptr) {
-    bool canRead = exec->canUseCollection(coll->name(), auth::Level::RO);
-    if (exec->databaseAuthLevel() == auth::Level::NONE || !canRead) {
-      return Result(TRI_ERROR_FORBIDDEN, "cannot access " + coll->name());
-    }
+  ExecContext const& exec = ExecContext::current();
+  bool canRead = exec.canUseCollection(coll->name(), auth::Level::RO);
+  if (exec.databaseAuthLevel() == auth::Level::NONE || !canRead) {
```

Comment:
IMHO we should switch the operands here, i.e.
```suggestion
  if (!canRead || exec.databaseAuthLevel() == auth::Level::NONE) {
```